numpy/ma/tests/test_core.py

```diff
@@ -1808,6 +1808,11 @@ def test_default_fill_value_structured(self):
         assert_equal(f1, expected)
         assert_equal(f2, expected)
 
+    def test_default_fill_value_void(self):
+        dt = np.dtype([('v', 'V7')])
```

Comment:
Indeed, that does fail in master - and `default_fill_value` is running those exact lines anyway.

It's not clear cut to me whether that should pass or fail, which is why I didn't include a test for it. It'd be better to test that `np.cancast(fill_value, dt)`, and make `default_fill_value` return numpy scalars